admob/integration_test/src/integration_test.cc

```diff
@@ -1195,6 +1269,201 @@ TEST_F(FirebaseAdMobTest, TestInterstitialAdErrorBadExtrasClassName) {
+  // kAdMobErrorLoadInProgress error.
+  firebase::admob::AdRequest request = GetAdRequest();
+  firebase::Future<firebase::admob::AdResult> first_load_ad =
+      rewarded->LoadAd(kRewardedAdUnit, request);
+  firebase::Future<firebase::admob::AdResult> second_load_ad =
+      rewarded->LoadAd(kRewardedAdUnit, request);
+
+  WaitForCompletion(second_load_ad, "Second LoadAd",
+                    firebase::admob::kAdMobErrorLoadInProgress);
+  WaitForCompletion(first_load_ad, "First LoadAd");
+
+  const firebase::admob::AdResult* result_ptr = second_load_ad.result();
+  ASSERT_NE(result_ptr, nullptr);
+  EXPECT_FALSE(result_ptr->is_successful());
+  EXPECT_EQ(result_ptr->code(), firebase::admob::kAdMobErrorLoadInProgress);
+  EXPECT_EQ(result_ptr->message(), "Ad is currently loading.");
+  EXPECT_EQ(result_ptr->domain(), "SDK");
+  const firebase::admob::ResponseInfo response_info =
+      result_ptr->response_info();
+  EXPECT_TRUE(response_info.adapter_responses().empty());
+  delete rewarded;
+}
+
+TEST_F(FirebaseAdMobTest, TestRewardedAdErrorBadAdUnitId) {
+  SKIP_TEST_ON_DESKTOP;
+  SKIP_TEST_ON_IOS;
+
+  firebase::admob::RewardedAd* rewarded = new firebase::admob::RewardedAd();
+  WaitForCompletion(rewarded->Initialize(app_framework::GetWindowContext()),
+                    "Initialize");
+
+  // Load the rewarded ad.
+  firebase::admob::AdRequest request = GetAdRequest();
+  firebase::Future<firebase::admob::AdResult> load_ad =
+      rewarded->LoadAd(kBadAdUnit, request);
+  WaitForCompletion(load_ad, "LoadAd",
+                    firebase::admob::kAdMobErrorInvalidRequest);
+
+  const firebase::admob::AdResult* result_ptr = load_ad.result();
+  ASSERT_NE(result_ptr, nullptr);
+  EXPECT_FALSE(result_ptr->is_successful());
+  EXPECT_EQ(result_ptr->code(), firebase::admob::kAdMobErrorInvalidRequest);
+  EXPECT_FALSE(result_ptr->message().empty());
+  EXPECT_EQ(result_ptr->domain(), kErrorDomain);
+  const firebase::admob::ResponseInfo response_info =
+      result_ptr->response_info();
+  EXPECT_TRUE(response_info.adapter_responses().empty());
+  delete rewarded;
+}
+
+TEST_F(FirebaseAdMobTest, TestRewardedAdErrorBadExtrasClassName) {
+  SKIP_TEST_ON_DESKTOP;
+  SKIP_TEST_ON_IOS;
+
+  firebase::admob::RewardedAd* rewarded = new firebase::admob::RewardedAd();
+  WaitForCompletion(rewarded->Initialize(app_framework::GetWindowContext()),
+                    "Initialize");
+
+  // Load the rewarded ad.
+  firebase::admob::AdRequest request = GetAdRequest();
+  request.add_extra(kAdNetworkExtrasInvalidClassName, "shouldnot", "work");
+  WaitForCompletion(rewarded->LoadAd(kRewardedAdUnit, request), "LoadAd",
+                    firebase::admob::kAdMobErrorAdNetworkClassLoadError);
+  delete rewarded;
+}
+
+// Stress tests.  These take a while so run them near the end.
+TEST_F(FirebaseAdMobTest, TestBannerViewStress) {
+  // AdMob cannot be tested on Firebase Test Lab, so disable tests on FTL.
+  TEST_REQUIRES_USER_INTERACTION;
```

Comment:
> AdMob cannot be tested on Firebase Test Lab, so disable tests on FTL.

What's the deal with this?

Also, even if we can't run this test on FTL, it would be better here to just ensure that you are specifically not running on FTL - the automated tests should still run on simulator, no?

Let's talk to @sunmou99 about seeing if there's a way to add a SKIP_TEST_ON_FIREBASE_TEST_LAB macro.

Reply:
I've enabled this test successfully in testing another PR, so I'll just remove it here.

I think that this TEST_REQUIRES_USER_INTERACTION was originally for the main BannerView tests that required you to click things but I was confused by the comment that was there, which seems to make it sound like the tests don't run at all.  I hadn't tested this until recentlly and it works fine.

Reply:
Done.